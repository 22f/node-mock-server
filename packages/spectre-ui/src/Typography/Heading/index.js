// @flow

import "spectre.css";
import React from "react";
import type { HeadingPropsType } from "../../../spectre-ui.js.flow";

const Heading = ({ tag, type = "h1", children, className = "" }: HeadingPropsType): React$Element<*> => {
	const Tag = `${tag || type}`;
	return <Tag className={`${className} ${type}`}>{children}</Tag>;
};

Heading.displayName = "Typography/Heading";

export default Heading;
