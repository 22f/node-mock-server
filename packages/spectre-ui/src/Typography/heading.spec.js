/* global describe it expect */

import React from "react";
import { shallow } from "enzyme";
import Heading from "./heading";

const props: Object = {
	children: <div />,
	type: "h2",
	tag: "h2"
};

describe("Heading", () => {
	describe("render", () => {
		it("snapshot: default", () => {
			const _props = { ...props };
			const tree = shallow(<Heading {..._props} />);
			expect(tree).toMatchSnapshot();
		});
		it("snapshot: no children", () => {
			const _props = { ...props, children: undefined };
			const tree = shallow(<Heading {..._props} />);
			expect(tree).toMatchSnapshot();
		});
		it("snapshot: no type", () => {
			const _props = { ...props, type: undefined };
			const tree = shallow(<Heading {..._props} />);
			expect(tree).toMatchSnapshot();
		});
		it("snapshot: no tag", () => {
			const _props = { ...props, tag: undefined };
			const tree = shallow(<Heading {..._props} />);
			expect(tree).toMatchSnapshot();
		});
		it("snapshot: no tag & no type", () => {
			const _props = { ...props, tag: undefined, type: undefined };
			const tree = shallow(<Heading {..._props} />);
			expect(tree).toMatchSnapshot();
		});
	});
});
