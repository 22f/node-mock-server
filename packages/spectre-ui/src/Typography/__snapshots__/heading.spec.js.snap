// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading render snapshot: default 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <h2
    className="h2"
>
    <div />
</h2>,
  "nodes": Array [
    <h2
      className="h2"
>
      <div />
</h2>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Heading
        tag="h2"
        type="h2"
>
        <div />
</Heading>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Heading {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <div />,
          "tag": "h2",
          "type": "h2",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <h2
          className="h2"
>
          <div />
</h2>,
        "_debugID": 2,
        "_renderedOutput": <h2
          className="h2"
>
          <div />
</h2>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Heading
    tag="h2"
    type="h2"
>
    <div />
</Heading>,
}
`;

exports[`Heading render snapshot: no children 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <h2
    className="h2"
/>,
  "nodes": Array [
    <h2
      className="h2"
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Heading
        tag="h2"
        type="h2"
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Heading {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": undefined,
          "tag": "h2",
          "type": "h2",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <h2
          className="h2"
/>,
        "_debugID": 4,
        "_renderedOutput": <h2
          className="h2"
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Heading
    tag="h2"
    type="h2"
/>,
}
`;

exports[`Heading render snapshot: no tag & no type 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <h1
    className="h1"
>
    <div />
</h1>,
  "nodes": Array [
    <h1
      className="h1"
>
      <div />
</h1>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Heading
        tag={undefined}
        type={undefined}
>
        <div />
</Heading>,
      "_debugID": 9,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Heading {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <div />,
          "tag": undefined,
          "type": undefined,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 5,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <h1
          className="h1"
>
          <div />
</h1>,
        "_debugID": 10,
        "_renderedOutput": <h1
          className="h1"
>
          <div />
</h1>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Heading
    tag={undefined}
    type={undefined}
>
    <div />
</Heading>,
}
`;

exports[`Heading render snapshot: no tag 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <h2
    className="h2"
>
    <div />
</h2>,
  "nodes": Array [
    <h2
      className="h2"
>
      <div />
</h2>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Heading
        tag={undefined}
        type="h2"
>
        <div />
</Heading>,
      "_debugID": 7,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Heading {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <div />,
          "tag": undefined,
          "type": "h2",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 4,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <h2
          className="h2"
>
          <div />
</h2>,
        "_debugID": 8,
        "_renderedOutput": <h2
          className="h2"
>
          <div />
</h2>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Heading
    tag={undefined}
    type="h2"
>
    <div />
</Heading>,
}
`;

exports[`Heading render snapshot: no type 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <h2
    className="h1"
>
    <div />
</h2>,
  "nodes": Array [
    <h2
      className="h1"
>
      <div />
</h2>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Heading
        tag="h2"
        type={undefined}
>
        <div />
</Heading>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Heading {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <div />,
          "tag": "h2",
          "type": undefined,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <h2
          className="h1"
>
          <div />
</h2>,
        "_debugID": 6,
        "_renderedOutput": <h2
          className="h1"
>
          <div />
</h2>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Heading
    tag="h2"
    type={undefined}
>
    <div />
</Heading>,
}
`;
